frontend/src/components/form/api/__tests__/ApiWrapper.spec.js

```diff
@@ -1,63 +1,45 @@
-// Libraries
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'
 import Vue from 'vue'
 import Vuetify from 'vuetify'
 import flushPromises from 'flush-promises'
 import { createLocalVue, shallowMount } from '@vue/test-utils'
 import veeValidatePlugin from '@/plugins/veeValidate'
 import ApiWrapper from '../ApiWrapper.vue'
-import { VForm, VBtn } from 'vuetify/lib'
+import { VBtn, VForm } from 'vuetify/lib'
 import { ValidationObserver } from 'vee-validate'
 
-jest.mock('lodash')
-const { cloneDeep } = jest.requireActual('lodash')
-
-/*
-jest.mock('vee-validate', () => ({
-  validate: jest.fn().mockResolvedValue({
-    valid: true,
-    errors: []
-  })
-})) */
-
-jest.useFakeTimers()
+const { cloneDeep } = await vi.importActual('lodash')
 
 Vue.use(Vuetify)
 Vue.use(veeValidatePlugin)
 let vuetify
 
-// creates a mock Promise which resolves within 100ms with value
-function mockPromiseResolving(value) {
-  return new Promise((resolve) => {
-    const timer = setTimeout(() => {
-      clearTimeout(timer)
-      resolve(value)
-    }, 100)
-  })
-}
+let debounce
+const resolveDebounce = () => debounce()
 
-// creates a mock Promise which rejects within 100ms with value
-/*
-function mockPromiseRejecting (value) {
-  return new Promise((resolve, reject) => {
-    const timer = setTimeout(() => {
-      clearTimeout(timer)
-      reject(value)
-    }, 100)
-  })
-} */
+vi.mock('lodash', async (importOriginal) => {
+  const lodash = await importOriginal()
+  return {
+    ...lodash,
+    debounce: (callback) => new Promise((resolve) => (debounce = resolve)).then(callback),
```

Comment:
Looks very funky... This implementation only supports one debounce at a time, because there is only one global `let debounce`, right? But I guess it is fine for now.

Reply:
yes, but why would you have 2 debounce before a user interaction/web request/response...?

Reply:
It is possible to write a test where e.g. the user triggers two separate debounced buttons, and check whether everything is executed in the expected order. I think this kind of setup is not supported by this mock.